src/AppInstallerCLITests/WorkFlow.cpp

```diff
@@ -1976,4 +1982,37 @@ TEST_CASE("InstallerWithoutDependencies_RootDependenciesAreUsed", "[dependencies
     // Verify root dependencies are shown
     REQUIRE(installOutput.str().find(Resource::LocString(Resource::String::InstallAndUpgradeCommandsReportDependencies).get()) != std::string::npos);
     REQUIRE(installOutput.str().find("PreviewIISOnRoot") != std::string::npos);
-}
\ No newline at end of file
+}
+
+TEST_CASE("OpenSource_WithCustomHeader", "[OpenSource][CustomHeader]")
+{
+    SetSetting(Streams::UserSources, R"(Sources:)"sv);
+    TestHook_ClearSourceFactoryOverrides();
+
+    SourceDetails details;
+    details.Name = "restsource";
+    details.Type = "Microsoft.Rest";
+    details.Arg = "thisIsTheArg";
+    details.Data = "thisIsTheData";
+    details.CustomHeader = "CustomHeader";
+
+    bool receivedCustomHeader = false;
+    TestSourceFactory factory{ TestRestSource::Create };
+    factory.OnAdd = [&](SourceDetails& sd) { receivedCustomHeader = details.CustomHeader.value().compare(sd.CustomHeader.value()) == 0; };

```

Comment:
>receivedCustomHeader [](http://example.com/codeflow?start=45&length=20)

Forgot to verify this after AddSource()?

Reply:
Just found out you already tested this in AddSource_CustomHeader,
These are just setup for OpenSource test